### DISCUSSION TITLE
Real-time - Fetching Duplicate Items

### DISCUSSION THREAD

**Discussion by babakfp** [pocketbase/pocketbase#2158] [Q&A] [answered]
Hi

I'm creating a real-time chat feature (same as Telegram or Discord). On the back-end side, I fetch the first 50 `messages`. Whenever users send a new message, I fetch it with the real-time feature. I store both data in a reactive variable, meaning data gets updated on the page whenever the value changes... Whenever users scroll to the top to see older messages, older messages get fetched and shown.

If the users don't send a new message and just scroll top to see the older messages, everything works fine. Let's say we get the first 50 messages, then we send/create 2 new messages, then we scroll to the top to see the older messages. Now, we are going to receive 2 of the older messages of that first/initial 50 messages again. So, this is the problem that I'm facing. I don't want to receive that 2 duplicate messages.

Video: https://drive.google.com/file/d/1N2wW2u6VVa8KkZcgtC2K6qhccvgZGQp3/view?usp=sharing

Any solutions?

Thanks

**Comment by ganigeorgiev** [comment, marked as answer]
If I'm understanding your use-case, the easiest solution would be to filter them on the client-side (_`?filter` and `?expand` support for the realtime events will be handled in #527 and #1657_) or when you update your list variable to skip the already existing records (we do something similar with the relations picker in the Admin UI).

Alternatively, you can also use a cursor-style pagination, eg.:

```js
await pb.collection("example").getList(1, 50, {
    filter: `created > "${someLastRecordCreatedDate}"` // reverse the operator based on your usecase
})
```

The datetime fields have a milliseconds part so the records matching should be granular enough for most use cases.
